src/shadertools/src/lib/inject-shader.js

```diff
@@ -0,0 +1,53 @@
+import {MODULE_INJECTORS_VS, MODULE_INJECTORS_FS} from '../modules/module-injectors';
+import {VERTEX_SHADER, FRAGMENT_SHADER} from './constants';
+
+const MODULE_INJECTORS = {
+  [VERTEX_SHADER]: MODULE_INJECTORS_VS,
+  [FRAGMENT_SHADER]: MODULE_INJECTORS_FS
+};
+
+// A minimal shader injection/templating system.
+// RFC: https://github.com/uber/luma.gl/blob/master/dev-docs/RFCs/v6.0/shader-injection-rfc.md
+export default function injectShader(source, type, inject, injectStandardStubs) {
+  const INJECT_DECLARATIONS = /^/;
+  const INJECT_CODE = /}[^{}]*$/;
```

Comment:
- Nit: Define constants outside this function
- Is it guaranteed that main() is the last in the shader code? Add a comment about the assumption here?